File: frontend/src/types/vocab.ts
export interface VocabItem {
  id: number
  word: string
  phonetic: string | null
  definition: string | null
  note: string | null
  dictionary_id: number | null
  created_at: string
}

export interface VocabListResponse {
  items: VocabItem[]
  total: number
  page: number
  page_size: number
}

File: backend/app/schemas/vocab.py
from datetime import datetime

from pydantic import BaseModel, Field


class VocabCreateRequest(BaseModel):
    word: str = Field(min_length=1, max_length=255)
    dictionary_id: int | None = None
    note: str | None = None


class VocabItemOut(BaseModel):
    id: int
    word: str
    phonetic: str | None
    definition: str | None
    note: str | None
    dictionary_id: int | None
    created_at: datetime

    model_config = {"from_attributes": True}


class VocabListResponse(BaseModel):
    items: list[VocabItemOut]
    total: int
    page: int
    page_size: int
